src/librustc/middle/region.rs

```diff
@@ -1429,6 +1429,9 @@ impl<'tcx> Visitor<'tcx> for RegionResolutionVisitor<'tcx> {
         self.terminating_scopes = outer_ts;
     }
 
+    fn visit_arg(&mut self, arg: &'tcx Arg) {
+        resolve_pat(self, &arg.pat);
+    }
```

Comment:
This should be unnecessary?
The default visitor already visits patterns in the arguments.

Reply:
It probably came in some of my attempts to make `unused_mut` work